Suggest using `SentryExceptionCaptured` instead of `WithSentry` (#11833)
getsentry/sentry-docs: platform-includes/getting-started-use/javascript.nestjs.mdx

Afterwards, add the `SentryModule` as a root module to your main module:

```javascript {filename: app.module.ts} {2, 8}
import { Module } from "@nestjs/common";
import { SentryModule } from "@sentry/nestjs/setup";
import { AppController } from "./app.controller";
import { AppService } from "./app.service";

@Module({
  imports: [
    SentryModule.forRoot(),
    // ...other modules
  ],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
```

> In case you are using a global catch-all exception filter (which is either a filter registered with `app.useGlobalFilters()` or a filter registered in your app module providers annotated with a `@Catch()` decorator without arguments), add a `@SentryExceptionCaptured()` decorator to the `catch()` method of this global error filter.
> This decorator will report all unexpected errors that are received by your global error filter to Sentry:

Reviewer: ```suggestion
If you're using a global catch-all exception filter (which is either a filter registered with `app.useGlobalFilters()` or a filter registered in your app module providers that's annotated with a `@Catch()` decorator without arguments), add a `@SentryExceptionCaptured()` decorator to the `catch()` method to your filter.
This decorator will then report all unexpected errors that are received by your global error filter to Sentry:
```

```javascript {2, 6}
import { Catch, ExceptionFilter } from '@nestjs/common';
import { SentryExceptionCaptured } from '@sentry/nestjs';

@Catch()
export class YourCatchAllExceptionFilter implements ExceptionFilter {
  @SentryExceptionCaptured()
  catch(exception, host): void {
    // your implementation here
  }
}
```

{/* TODO(v9): Remove this note */}
_Note that `@SentryExceptionCaptured()` was called `@WithSentry` in SDK versions `8.38.0` and prior._

> In case you do not have a global catch-all exception filter, add the `SentryGlobalFilter` to the providers of your main
> module. This filter will report all unhandled errors to Sentry that are not caught by any other error filter.
> **Important:** The `SentryGlobalFilter` needs to be registered before any other exception filters.

Reviewer: ```suggestion
If you don't have a global catch-all exception filter, add the `SentryGlobalFilter` to the providers of your main
module. This filter will then report any unhandled errors that aren't caught by other error filters to Sentry.
**Important:** The `SentryGlobalFilter` needs to be registered before any other exception filters.
```
